modules/electrophysiology_uploader/php/upload.class.inc

```diff
@@ -0,0 +1,249 @@
+    {
+        // Ensure GET or POST request.
+        switch ($request->getMethod()) {
+        case 'GET':
+            return $this->_handleGet($request);
+        case 'POST':
+            return $this->_handlePOST($request);
+        default:
+            return new \LORIS\Http\Response\JSON\MethodNotAllowed(
+                $this->allowedMethods()
+            );
+        }
+    }
+
+    /**
+     * Handle an incoming HTTP GET request to download uploaded file.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface
+     */
+    private function _handleGet(ServerRequestInterface $request) : ResponseInterface
+    {
+        $qParams = $request->getQueryParams();
+        if (!isset($qParams['upload_id'])) {
+            return new \LORIS\Http\Response\JSON\BadRequest(
+                "upload_id query parameter is required"
+            );
+        }
+
+        $db   = $this->loris->getDatabaseConnection();
+        $file = $db->pselectOne(
+            "
+            SELECT UploadLocation
+            FROM electrophysiology_uploader
+            WHERE UploadID = :uploadID
+        ",
+            ['uploadID' => $qParams['upload_id']]
+        );
+
+        if (empty($file)) {
+            return new \LORIS\Http\Response\JSON\NotFound(
+                "File not found."
+            );
+        }
+
+        $filename      = urldecode(basename($file));
+        $pos           = strrpos($file, '/');
+        $file_rel_path = $pos === false ? '' : substr($file, 0, $pos);
+        $config        = \NDB_Config::singleton();
+        $upload_dir    = new \SplFileInfo(
+            $config->getSetting('EEGUploadIncomingPath')
+        );
+
+        $downloader = new \LORIS\FilesDownloadHandler(
+            new \SPLFileInfo($upload_dir . '/' . $file_rel_path)
+        );
+
+        return $downloader->handle(
+            $request->withAttribute('filename', $filename)
+        );
+    }
+
+    /**
+     * Processes the values & saves to database and return a json response.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request.
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handlePOST(ServerRequestInterface $request) : ResponseInterface
+    {
+        $db = $this->loris->getDatabaseConnection();
+
+        $uploadedFile = $request->getUploadedFiles()['eegFile'] ?? null;
+        $values       = $request->getParsedBody();
+        $pscid        = $values['pscid'] ?? null;
+        $candID       = $values['candID'] ?? null;
+        $visitLabel   = $values['visit'] ?? null;
+
+        if (empty($request->getUploadedFiles()) && empty($values)) {
+            $upload_max_size = \Utility::getMaxUploadSize();
+            return new \LORIS\Http\Response\JSON\InternalServerError(
+                "The uploaded file size ($upload_max_size) exceeds " .
```

Comment:
This error message is wrong. The variable contains the max upload size, not the size of the uploaded file.